backend/api/EventHandlers/MqttEventHandler.cs

```diff
@@ -55,33 +60,161 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             await stoppingToken;
         }
 
-        private async void OnIsarConnect(object? sender, MqttReceivedArgs mqttArgs)
+        private async void OnIsarRobotStatus(object? sender, MqttReceivedArgs mqttArgs)
         {
-            var isarRobot = (IsarConnectMessage)mqttArgs.Message;
-            var robot = await RobotService.ReadByName(isarRobot.RobotId);
+            var isarRobotStatus = (IsarRobotStatusMessage)mqttArgs.Message;
+            var robot = await RobotService.ReadByName(isarRobotStatus.RobotName);
+
             if (robot == null)
             {
-                _logger.LogError(
-                    "ISAR instance for robot {name} connected. Could not find robot {name} in the database.",
-                    isarRobot.RobotId,
-                    isarRobot.RobotId
+                _logger.LogInformation(
+                    "Received message from unknown ISAR instance with robot name {name}.",
+                    isarRobotStatus.RobotName
                 );
                 return;
             }
+
+            robot.Status = isarRobotStatus.RobotStatus;
+            robot = await RobotService.Update(robot);
+            _logger.LogInformation(
+                "Updated status for robot {name} to {status}",
+                robot.Name,
+                robot.Status
+            );
+
+            if (_isarConnectionTimers.ContainsKey(isarRobotStatus.RobotName))
+            {
+                _logger.LogInformation(
+                    "Reset connection timer for robot {name}",
+                    isarRobotStatus.RobotName
+                );
+                _isarConnectionTimers[isarRobotStatus.RobotName].Reset();
```

Comment:
Are we certain that "stop" and "start" of timer resets the timer's clock?

Reply:
Fairly certain yes. I found it slightly counterintuitive too but some googling told me that was the way to solve it. 

And it works in practise.

Reply:
Ok, sounds good